app/components/billing-invoices.js

```diff
@@ -0,0 +1,25 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+
+  invoiceYears: computed('invoices.@each.{createdAt}', function () {
+    const invoiceYears = this.invoices.map(invoice => invoice.createdAt.getFullYear());
+    const distinctInvoiceYears = new Set(invoiceYears);
+    const sortedDistinctInvoiceYears = [...distinctInvoiceYears].sort((a, b) => b - a);
+    return sortedDistinctInvoiceYears;
+  }),
+
+  year: computed('invoiceYears.[]', {
+    get() {
+      return this.invoiceYears.get('firstObject');
+    },
+    set(key, value) {
+      return value;
+    }
+  }),
+
+  selectedInvoices: computed('invoices.@each.{createdAt}', 'year', function () {
+    return this.invoices.filter(invoice => invoice.createdAt.getFullYear() === this.year);
```

Comment:
To simplify things here I'd recommend to add a computed property `year` to [Invoice model](https://github.com/travis-ci/travis-web/blob/master/app/models/invoice.js):
```js
year: computed('createdAt', function () {
  return this.createdAt.getFullYear();
})
```

After that you'd be able to refer it here instead of calculating manually. E.g. `invoiceYears` would be just:
```js
return this.invoices.mapBy('year').uniq().sort((a, b) => b - a);
```
And here it would be just:
```js
return this.invoices.filterBy('year', this.year);
```